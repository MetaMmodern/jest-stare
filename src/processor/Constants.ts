import * as chalk from "chalk";

/**
 * Contains Constants constants
 * @export
 * @class Constants
 */
export class Constants {

    /**
     * Name (same as package.json)
     * @static
     * @memberof Constants
     */
    public static readonly NAME = "jest-stare";

    /**
     * Logo for messaging
     * @static
     * @memberof Constants
     */
    public static readonly LOGO = chalk.default.green("**  ") + chalk.default.green("jest") + chalk.default.yellow("-") + chalk.default.red("stare");

    /**
     * Default jest-stare results directory
     * @static
     * @memberof Constants
     */
    public static readonly DEFAULT_RESULTS_DIR = "./" + Constants.NAME;

    /**
     * Main HTML index file for report
     * @static
     * @memberof Constants
     */
    public static readonly MAIN_HTML = "index.html";

    /**
     * jest-stare render JS file
     * @static
     * @memberof Constants
     */
    public static readonly JEST_STARE_JS = "view.js";

    /**
     * Raw JSON response results
     * @static
     * @memberof Constants
     */
    public static readonly RESULTS_RAW = "jest-results.json";

    /**
     * jest-stare custom css
     * @static
     * @memberof Constants
     */
    public static readonly JEST_STARE_CSS = Constants.NAME + ".css";

    /**
     * Template HTML file
     * @static
     * @memberof Constants
     */
    public static readonly TEMPLATE_HTML = "template.html";

    /**
     * Base dir to contain css files
     * @static
     * @memberof Constants
     */
    public static readonly CSS_DIR = "css/";

    /**
     * Base dir to contain js files
     * @static
     * @memberof Constants
     */
    public static readonly JS_DIR = "js/";

    /**
     * Log content after logo
     * @static
     * @memberof Constants
     */
    public static readonly LOG_MESSAGE = " --testResultsProcessor: wrote output report to ";
}
