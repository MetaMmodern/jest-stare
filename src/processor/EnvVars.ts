import { IJestStareConfig } from "./doc/IJestStareConfig";
import { Logger } from "../utils/Logger";
import { EnvVarService } from "../utils/EnvVarService";

/**
 * Service for reading environmental variables
 * @export
 * @class EnvVars
 */
export class EnvVars {

    /**
     * Prefix for all jest-stare related environmental variables
     * @static
     * @type {string}
     * @memberof EnvVars
     */
    public static readonly ENV_PREFIX: string = "JEST_STARE_";

    /**
     * Creates an instance of EnvVars.
     * @memberof EnvVars
     */
    constructor(private mEnvSrv = new EnvVarService(EnvVars.ENV_PREFIX)) {}

    /**
     * Read all jest-stare related environmental variables
     * @returns {IJestStareConfig} - jest stare config from env vars
     * @memberof EnvVars
     */
    public read(): IJestStareConfig {
        const additionalResultsProcessorsValue = this.mEnvSrv.readEnvValue("ADDITIONAL_RESULTS_PROCESSORS");
        let additionalResultsProcessors: string[];
        if (additionalResultsProcessorsValue != null) {
            try {
                additionalResultsProcessors = JSON.parse(additionalResultsProcessorsValue);
            } catch (e) {
                Logger.get.error("Could not parse additional results processors value." +
                    "It should be a JSON string of an array of strings, like the following: " +
                    JSON.stringify(["jest-html-reporter"]) + "\n You specified: " + additionalResultsProcessorsValue);
            }
        }
        return {
            resultDir: this.mEnvSrv.readEnvValue("RESULT_DIR"),
            resultJson: this.mEnvSrv.readEnvValue("RESULT_JSON"),
            resultHtml: this.mEnvSrv.readEnvValue("RESULT_HTML"),
            log: this.mEnvSrv.readBoolEnvValue("LOG"),
            merge: this.mEnvSrv.readBoolEnvValue("MERGE"),
            jestStareConfigJson: this.mEnvSrv.readEnvValue("CONFIG_JSON"),
            coverageLink: this.mEnvSrv.readEnvValue("COVERAGE_LINK"),
            additionalResultsProcessors
        };
    }

    /**
     * Combine jest stare config from package json and environmental variables,
     * with environmental variables taking precedence
     * @param packageJsonConfig
     * @param {IJestStareConfig} packageJsonConfig - package json config
     * @param {IJestStareConfig} envConfig - env based config
     * @returns {IJestStareConfig} - merged config
     * @memberof EnvVars
     */
    public resolve(packageJsonConfig: IJestStareConfig, envConfig: IJestStareConfig): IJestStareConfig {
        const mergedConfig: IJestStareConfig = {};

        if (envConfig.resultDir != null || packageJsonConfig.resultDir != null) {
            mergedConfig.resultDir = envConfig.resultDir == null ? packageJsonConfig.resultDir : envConfig.resultDir;
        }

        if (envConfig.resultJson != null || packageJsonConfig.resultJson != null) {
            mergedConfig.resultJson = envConfig.resultJson == null ? packageJsonConfig.resultJson : envConfig.resultJson;
        }

        if (envConfig.resultHtml != null || packageJsonConfig.resultHtml != null) {
            mergedConfig.resultHtml = envConfig.resultHtml == null ? packageJsonConfig.resultHtml : envConfig.resultHtml;
        }

        if (envConfig.log != null || packageJsonConfig.log != null) {
            mergedConfig.log = envConfig.log == null ? packageJsonConfig.log : envConfig.log;
        }

        if (envConfig.merge != null || packageJsonConfig.merge != null) {
            mergedConfig.merge = envConfig.merge == null ? packageJsonConfig.merge : envConfig.merge;
        }

        if (envConfig.jestStareConfigJson != null || packageJsonConfig.jestStareConfigJson != null) {
            mergedConfig.jestStareConfigJson =
                envConfig.jestStareConfigJson == null ? packageJsonConfig.jestStareConfigJson : envConfig.jestStareConfigJson;
        }


        if (envConfig.coverageLink != null || packageJsonConfig.coverageLink != null) {
            mergedConfig.coverageLink = envConfig.coverageLink == null ? packageJsonConfig.coverageLink : envConfig.coverageLink;
        }

        if (envConfig.additionalResultsProcessors != null || packageJsonConfig.additionalResultsProcessors != null) {
            mergedConfig.additionalResultsProcessors =
                envConfig.additionalResultsProcessors == null ? packageJsonConfig.additionalResultsProcessors : envConfig.additionalResultsProcessors;
        }

        return mergedConfig;
    }
}
