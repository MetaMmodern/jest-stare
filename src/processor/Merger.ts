import { IResultsProcessorInput } from "./doc/jest/IResultsProcessorInput";
import { ITestResults } from "./doc/jest/ITestResults";

/**
 * Merges old and new test results
 * @export
 * @class Merger
 */
export class Merger {

    /**
     * Combine test results
     * @param {IResultsProcessorInput} oldResults - old results
     * @param {IResultsProcessorInput} newResults - newer results
     * @returns {IResultsProcessorInput} - combined results
     * @memberof Merger
     */
    public combine(oldResults: IResultsProcessorInput, newResults: IResultsProcessorInput): IResultsProcessorInput {
        const combinedResults = newResults;
        const dupMap = new Map<string, ITestResults>();

        // save old test results
        oldResults.testResults.forEach( (testResult) => {
            dupMap.set(testResult.testFilePath, testResult);
        });

        // save only new test results (uniqueness by filepath)
        newResults.testResults.forEach( (testResult) => {
            if (!dupMap.has(testResult.testFilePath)) {
                dupMap.set(testResult.testFilePath, testResult);
            }
        });

        // delete everything
        combinedResults.testResults = [];

        dupMap.forEach((value, key) => {
            combinedResults.testResults.push(value);
        });

        return combinedResults;
    }
}
