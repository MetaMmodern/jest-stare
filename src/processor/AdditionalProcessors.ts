import { IResultsProcessorInput } from "./doc/jest/IResultsProcessorInput";
import { Logger } from "../utils/Logger";
import { Constants } from "./Constants";


export class AdditionalProcessors {

    /**
     * Pass the result processor input given to jest-stare to additional
     * test results processors
     * @param jestTestData - input passed to jest-stare
     * @param processors - list of test results processors (e.g. ["jest-html-reporter"])
     *                     to forward the data to
     */
    public static execute(jestTestData: IResultsProcessorInput, processors: string[]): void {
        const chalk = require("chalk");
        for (const processor of processors) {
            try {
                require(processor)(jestTestData);
                AdditionalProcessors.logger.debug(Constants.LOGO + " passed results to additional processor \"" +
                    processor + "\"" + chalk.default.green("\t**"));
            } catch (e) {
                AdditionalProcessors.logger.error("Error executing additional processor: \"" + processor + "\"" + e);
            }
        }
    }

    private static logger = new Logger();
}
