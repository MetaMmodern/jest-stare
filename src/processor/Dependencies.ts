import { IThirdPartyDependency } from "./doc/IThirdPartyDependency";
import { Constants } from "./Constants";

/**
 * Describe all third party dependencies - note, these must still be added to template manually for now
 * @export
 * @class Dependencies
 */
export class Dependencies {

    /**
     * All third party dependencies are here
     * @static
     * @type {IThirdPartyDependency[]}
     * @memberof Dependencies
     */
    public static readonly THIRD_PARTY_DEPENDENCIES: IThirdPartyDependency[] = [

        // js
        {
            requireDir: "bootstrap/dist/js/",
            file: "bootstrap.min.js",
            targetDir: Constants.JS_DIR
        },
        {
            requireDir: "diff2html/dist/",
            file: "diff2html.min.js",
            targetDir: Constants.JS_DIR
        },
        {
            requireDir: "jquery/dist/",
            file: "jquery.min.js",
            targetDir: Constants.JS_DIR
        },
        {
            requireDir: "holderjs/",
            file: "holder.js",
            targetDir: Constants.JS_DIR
        },

        // css
        {
            requireDir: "bootstrap/dist/css/",
            file: "bootstrap.min.css",
            targetDir: Constants.CSS_DIR
        },
        {
            requireDir: "diff2html/dist/",
            file: "diff2html.min.css",
            targetDir: Constants.CSS_DIR
        },
    ];
}
