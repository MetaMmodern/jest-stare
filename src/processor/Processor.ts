import { Constants } from "./Constants";
import { IResultsProcessorInput } from "./doc/jest/IResultsProcessorInput";
import { ISubstitute } from "./doc/ISubstitute";
import { IO } from "../utils/IO";
import * as mustache from "mustache";
import * as path from "path";
import { IJestStareConfig, PACKAGE_JSON_KEY } from "./doc/IJestStareConfig";
import { Logger } from "../utils/Logger";
import * as chalk from "chalk";
import { IThirdPartyDependency } from "./doc/IThirdPartyDependency";
import { Dependencies } from "./Dependencies";
import { isNullOrUndefined } from "util";
import { IProcessParms } from "./doc/IProcessParms";
import { EnvironmentalVariables } from "../utils/EnvironmentalVariables";
const pkgUp = require("pkg-up");

/**
 * Class to post process jest output and summarize information in an html file
 * @export
 * @class Processor
 */
export class Processor {

    /**
     * Main exported method to obtain and return summary results
     * @static
     * @param {IResultsProcessorInput} results - input results object
     * @param {IJestStareConfig} [explicitConfig] - programmatic config
     * @param {IProcessParms} [mProcessParms] - parms object to control process behavior
     * @returns - returns input results object
     * @memberof Processor
     */
    public static run(results: IResultsProcessorInput, explicitConfig?: IJestStareConfig,
                      parms?: IProcessParms) {

        return new Processor(results, explicitConfig, parms).generate();
    }

    /**
     * Instance of our logger
     * @private
     * @type {Logger}
     * @memberof Processor
     */
    private mLog: Logger;

    /**
     * Creates an instance of Processor.
     * @param {IResultsProcessorInput} results - input results object
     * @param {IJestStareConfig} [explicitConfig] - programmatic config
     * @param {IProcessParms} [mProcessParms] - parms object to control process behavior
     * @memberof Processor
     */
    constructor(private mResults: IResultsProcessorInput, private mExplicitConfig?: IJestStareConfig,
                private mProcessParms?: IProcessParms) {
    }

    /**
     * Generate a report after constructed
     * @private
     * @returns
     * @memberof Processor
     */
    private generate() {
        const substitute: ISubstitute = {};

        // throw error if no input object
        if (isNullOrUndefined(this.mResults)) {
            throw new Error(Constants.NO_INPUT);
        }

        const config = this.buildConfig();

        // build mustache render substitution values
        substitute.results = this.mResults;
        substitute.rawResults = JSON.stringify(this.mResults, null, 2);
        substitute.jestStareConfig = config;
        substitute.rawJestStareConfig = JSON.stringify(config, null, 2);

        // save in reporter
        if (this.mProcessParms && this.mProcessParms.reporter) {
            this.mProcessParms.reporter.jestStareConfig = config;
        }

        // generate report
        this.generateReport(config.resultDir, substitute, this.mProcessParms);

        if (config.additionalResultsProcessors != null) {
            this.execute(this.mResults, config.additionalResultsProcessors);
        }
        // return back to jest
        return this.mResults;
    }

    /**
     * Build config from explicit config, package.json, and defaults
     * @private
     * @returns {IJestStareConfig} - constructed config
     * @memberof Processor
     */
    private buildConfig(): IJestStareConfig {

        // get configuration
        const packageJsonConfig = this.readPackageJson();

        // read environmental variables and merge them with the package.json config (env takes precedence)
        const envVars = new EnvironmentalVariables();
        const mergedEnvAndPackageJsonConfig = envVars.resolve(packageJsonConfig, envVars.read());

        // explicit config takes precedence over  env and package.json
        const config = this.mExplicitConfig || mergedEnvAndPackageJsonConfig;

        // take packagejson options after setting explicit config (concatenate both)
        if (this.mExplicitConfig != null) {
            Object.keys(mergedEnvAndPackageJsonConfig).forEach((key) => {
                if (isNullOrUndefined(this.mExplicitConfig[key]) && !isNullOrUndefined(mergedEnvAndPackageJsonConfig[key])) {
                    config[key] = mergedEnvAndPackageJsonConfig[key];
                }
            });
        }

        if (config.resultDir == null) {
            config.resultDir = Constants.DEFAULT_RESULTS_DIR;
        }

        // suppress logging if requested
        // NOTE(Kelosky): must be first, to suppress all logging
        if (!isNullOrUndefined(config.log)) {
            if (!config.log) {
                this.logger.on = false;
            }
        }

        // record if we were invoked programmatically
        // NOTE(Kelosky): should be second, to record if override config
        if (!isNullOrUndefined(this.mExplicitConfig)) {

            // display if not internal invocation
            if (this.mProcessParms && this.mProcessParms.reporter) {
                // do nothing
            } else {
                this.logger.info(Constants.OVERRIDE_JEST_STARE_CONFIG);
            }
        }

        if (isNullOrUndefined(config.resultHtml)) {
            this.logger.debug("Setting to default resultHtml");
            config.resultHtml = Constants.MAIN_HTML;
        } else {
            if (config.resultHtml.indexOf( Constants.HTML_EXTENSION) === -1){
                // add .html if the user did not specify it
                config.resultHtml = config.resultHtml + Constants.HTML_EXTENSION;
            }
        }

        if (isNullOrUndefined(config.resultJson)) {
            config.resultJson = Constants.RESULTS_RAW;
        }

        return config;
    }

    /**
     * Create HTML report
     * @private
     * @param {string} resultDir -  directory to save report
     * @param {ISubstitute} substitute - substitution values for mustache render
     * @param {ISettings} settings - settings for IO
     * @memberof Processor
     */
    private generateReport(resultDir: string, substitute: ISubstitute, parms: IProcessParms) {

        // create base html file
        resultDir = resultDir + "/"; // append an extra slash in case the user didn't add one
        IO.mkdirsSync(resultDir);
        IO.writeFileSync(resultDir + substitute.jestStareConfig.resultHtml,
            mustache.render(this.obtainWebFile(Constants.TEMPLATE_HTML), substitute));

        // create raw json
        IO.writeFileSync(resultDir + substitute.jestStareConfig.resultJson, substitute.rawResults);

        // create jest-stare config if requested
        if (!isNullOrUndefined(substitute.jestStareConfig.jestStareConfigJson)) {
            IO.writeFileSync(resultDir + substitute.jestStareConfig.jestStareConfigJson, substitute.rawJestStareConfig);
        }

        // create our css
        const cssDir = resultDir + Constants.CSS_DIR;
        IO.mkdirsSync(cssDir);
        IO.writeFileSync(cssDir + Constants.JEST_STARE_CSS, this.obtainWebFile(Constants.JEST_STARE_CSS));

        // create our js
        const jsDir = resultDir + Constants.JS_DIR;
        IO.mkdirsSync(jsDir);
        IO.writeFileSync(jsDir + Constants.JEST_STARE_JS, this.obtainJsRenderFile(Constants.JEST_STARE_JS));

        // add third party dependencies
        Dependencies.THIRD_PARTY_DEPENDENCIES.forEach((dependency) => {
            // dependency.targetDir = resultDir + dependency.targetDir;
            const updatedDependency = Object.assign({}, ...[dependency]);
            updatedDependency.targetDir = resultDir + dependency.targetDir;
            this.addThirdParty(updatedDependency);
        });

        // log complete
        let type = " ";
        type += (parms && parms.reporter) ? Constants.REPORTERS : Constants.TEST_RESULTS_PROCESSOR;
        this.logger.info(Constants.LOGO + type + Constants.LOG_MESSAGE + resultDir + substitute.jestStareConfig.resultHtml + Constants.SUFFIX);
    }

    /**
     * Pass the result processor input given to jest-stare to additional
     * test results processors
     * @param jestTestData - input passed to jest-stare
     * @param {IResultsProcessorInput} jestTestData - input passed to jest-stare
     * @param {string[]} processors - processors
     * @param processors - list of test results processors (e.g. ["jest-html-reporter"])
     *                     to forward the data to
     * @memberof Processor
     */
    private execute(jestTestData: IResultsProcessorInput, processors: string[]): void {
        for (const processor of processors) {
            if (processor === Constants.NAME) {
                this.logger.error("Error: In order to avoid infinite loops, " +
                    "jest-stare cannot be listed as an additional processor. Skipping... ");
                continue;
            }
            try {
                require(processor)(jestTestData);
                this.logger.info(Constants.LOGO + " passed results to additional processor " +
                    chalk.default.white("\"" + processor + "\"") + Constants.SUFFIX);
            } catch (e) {
                this.logger.error("Error executing additional processor: \"" + processor + "\" " + e);
            }
        }
    }

    /**
     * Add all third party dependencies
     * @private
     * @param {IThirdPartyDependency} dependency - a dependency to add
     * @memberof Processor
     */
    private async addThirdParty(dependency: IThirdPartyDependency) {
        const location = require.resolve(dependency.requireDir + dependency.file);
        await IO.writeFileSync(dependency.targetDir + dependency.file, IO.readFileSync(location));
    }

    /**
     * Obtain web files
     * @private
     * @returns {string} - file contents from web directory
     * @memberof Processor
     */
    private obtainWebFile(name: string): string {
        return IO.readFileSync(path.resolve(__dirname + "/../../web/" + name));
    }

    /**
     * Obtain js files
     * @private
     * @returns {string} - js file contents from js directory
     * @memberof Processor
     */
    private obtainJsRenderFile(name: string): string {
        return IO.readFileSync(path.resolve(__dirname + "/../render/" + name));
    }

    /**
     * Read from the user's package.json, if present
     * @private
     * @returns {IJestStareConfig} - config object
     * @memberof Processor
     */
    private readPackageJson(): IJestStareConfig {
        const packageJson = pkgUp.sync();
        if (packageJson !== null) {
            const packageJsonContents = IO.readFileSync(packageJson).toString();
            const packageJsonObject = JSON.parse(packageJsonContents);
            if (packageJsonObject[PACKAGE_JSON_KEY] == null) {
                // package json found, but no jest stare config
                return {};
            } else {
                // found the user's package.json config
                return packageJsonObject[PACKAGE_JSON_KEY];
            }
        } else {
            // if we can't find any package.json, return a blank config
            return {};
        }

    }

    /**
     * Set logger instance
     * @private
     * @memberof Processor
     */
    set logger(logger: Logger) {
        this.mLog = logger;
    }

    /**
     * Get log instance
     * @readonly
     * @private
     * @memberof Processor
     */
    get logger() {
        if (isNullOrUndefined(this.mLog)) {
            this.logger = new Logger();
        }
        // console.log((this.mLog as any).writeStdout)
        return this.mLog;
    }
}
