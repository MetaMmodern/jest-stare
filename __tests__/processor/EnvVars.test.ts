import { EnvVars } from "../../src/processor/EnvVars";

describe("EnvVars tests", () => {
    it("should have an empty config for no env variables set", () => {
        const response = new EnvVars().read();
        expect(response).toMatchSnapshot();
    });

    it("should have a complete config for all env variables set", () => {
        const JEST_STARE_RESULT_DIR = "JEST_STARE_RESULT_DIR";
        const JEST_STARE_RESULT_JSON = "JEST_STARE_RESULT_JSON";
        const JEST_STARE_RESULT_HTML = "JEST_STARE_RESULT_HTML";
        const JEST_STARE_LOG = "JEST_STARE_LOG";
        const JEST_STARE_MERGE = "JEST_STARE_MERGE";
        const JEST_STARE_CONFIG_JSON = "JEST_STARE_CONFIG_JSON";
        const JEST_STARE_COVERAGE_LINK = "JEST_STARE_COVERAGE_LINK";
        const JEST_STARE_ADDITIONAL_RESULTS_PROCESSORS = "JEST_STARE_ADDITIONAL_RESULTS_PROCESSORS";

        process.env[JEST_STARE_RESULT_DIR] = "some/dir";
        process.env[JEST_STARE_RESULT_JSON] = "some.json";
        process.env[JEST_STARE_RESULT_HTML] = "some.html";
        process.env[JEST_STARE_LOG] = "false";
        process.env[JEST_STARE_MERGE] = "1";
        process.env[JEST_STARE_CONFIG_JSON] = "true";
        process.env[JEST_STARE_COVERAGE_LINK] = "www.walmart.com";
        process.env[JEST_STARE_ADDITIONAL_RESULTS_PROCESSORS] = "[\"one\", \"two\", \"three\"]";
        const response = new EnvVars().read();
        expect(response).toMatchSnapshot();
    });

    // it("should throw error when using jest-stare as an additional processor", () => {
    //     const JEST_STARE_ADDITIONAL_RESULTS_PROCESSORS = "JEST_STARE_ADDITIONAL_RESULTS_PROCESSORS";

    //     process.env[JEST_STARE_ADDITIONAL_RESULTS_PROCESSORS] = "bad json";

    //     let error;
    //     try {
    //         const response = new EnvVars().read();
    //     } catch (thrownError) {
    //         error = thrownError;
    //     }
    //     expect(error).toBeDefined();
    //     expect(error.msg).toMatchSnapshot();
    // });
});
