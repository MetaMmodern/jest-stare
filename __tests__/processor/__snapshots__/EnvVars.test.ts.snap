// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnvVars tests should have a complete config for all env variables set 1`] = `
Object {
  "additionalResultsProcessors": Array [
    "one",
    "two",
    "three",
  ],
  "coverageLink": "www.walmart.com",
  "disableCharts": false,
  "jestGlobalConfigJson": undefined,
  "jestStareConfigJson": "true",
  "log": false,
  "report": undefined,
  "resultDir": "some/dir",
  "resultHtml": "some.html",
  "resultJson": "some.json",
}
`;

exports[`EnvVars tests should have an empty config for no env variables set 1`] = `
Object {
  "additionalResultsProcessors": undefined,
  "coverageLink": undefined,
  "disableCharts": undefined,
  "jestGlobalConfigJson": undefined,
  "jestStareConfigJson": undefined,
  "log": undefined,
  "report": undefined,
  "resultDir": undefined,
  "resultHtml": undefined,
  "resultJson": undefined,
}
`;
